service/surf/apps/materials/utils.py

```diff
@@ -64,21 +64,30 @@ def add_extra_parameters_to_materials(user, materials):
     :param materials: array of materials
     :return: updated array of materials
     """
-    # perhaps this could be optimized by querying the database once for a list of materials
-    # however, we still want to add the 0 values for items we don't have in the database yet so this is easy for now
+
+    material_objects = list(Material.objects.filter(
+        external_id__in=(m["external_id"] for m in materials)
+    ))
+
+    educational_level_filters = MpttFilterItem.objects.filter(
+        name__in=(level for material in materials for level in material["educationallevels"])
+    ).distinct().select_related("title_translations").get_cached_trees()
```

Comment:
Je hebt hier de tree niet nodig volgens mij. Scheelt weer flink wat tijd, die de library nodig heeft om de boom uit te rekenen. Denk dat dit met een "filter" sowieso weinig zin heeft, want de children komen dan sowieso niet mee in de resultaten. Zonde van de moeite dus.

Reply:
Verder lijkt een dict me hier ook handig. Dan krijg je:

```
educational_level_filters = {
    filter.external_id: filter
    for filter in MpttFilterItem.objects.filter(name__in=(level for material in materials for level in material["educationallevels"])).distinct().select_related("title_translations")
}
```

Expres met external_id want name is niet betrouwbaar